sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -62,7 +62,7 @@ private[sql] object Dataset {
   def ofRows(sparkSession: SparkSession, logicalPlan: LogicalPlan): DataFrame = {
     val qe = sparkSession.sessionState.executePlan(logicalPlan)
     qe.assertAnalyzed()
-    new Dataset[Row](sparkSession, logicalPlan, RowEncoder(qe.analyzed.schema))
+    new Dataset[Row](sparkSession, qe, RowEncoder(qe.analyzed.schema), skipAnalysis = true)
```

Comment:
how about we remove the `qe.assertAnalyzed()` in `ofRows`? Then we don't need the `skipAnalysis` flag.